[FSDP][Easy] Fix return in docstrings

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #75431 [FSDP] Add `ignored_modules` ctor arg
* #75430 [FSDP] Fix `_get_param_to_unflat_param_names()` for shared params
* **#75327 [FSDP][Easy] Fix return in docstrings**

- This fixes the return docstrings for optim state checkpointing, following https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html as described in the contributing guide.
    - The return value name should not be specified, and there should not be an additional indent for lines after the first.
- I am unable to fix my docs rendering locally, so I am opening the PR directly.
- We can normalize the documentation for the rest of FSDP in later follow-ups.

Differential Revision: [D35431794](https://our.internmc.facebook.com/intern/diff/D35431794)

Commit:
[FSDP][Easy] Fix return in docstrings

[ghstack-poisoned]

awgu commented:
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431794).

Commit:
Update on "[FSDP][Easy] Fix return in docstrings"


Differential Revision: [D35431794](https://our.internmc.facebook.com/intern/diff/D35431794)

[ghstack-poisoned]

awgu commented:
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431794).

mrshenli commented on torch/distributed/fsdp/optim_utils.py:
```diff
@@ -89,13 +88,13 @@ def _communicate_optim_state(
-        flat_param (FlatParameter): The flattened parameter.
-        flat_param_state (Dict[str, Any]): The entry in the "state" part of
-        the optimizer state dict corresponding to the flattened parameter.
+        flat_param (FlatParameter): The flattened parameter. flat_param_state
+        (Dict[str, Any]): The entry in the "state" part of the optimizer state
+        dict corresponding to the flattened parameter.
```
Will this still display correctly after merging multiple args into the same line? I assume no?

awgu replied:
Good catch! I changed that argument by accident (meant to only change returns).

Commit:
Update on "[FSDP][Easy] Fix return in docstrings"


Fixing the return docstrings for optim state checkpointing, following https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html as described in the contributing guide.

I am unable to fix my local docs rendering, so I am opening the PR directly.

Differential Revision: [D35431794](https://our.internmc.facebook.com/intern/diff/D35431794)

[ghstack-poisoned]

Commit:
Update on "[FSDP][Easy] Fix return in docstrings"


- This fixes the return docstrings for optim state checkpointing, following https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html as described in the contributing guide.
    - The return value name should not be specified, and there should not be an additional indent for lines after the first.
- I am unable to fix my docs rendering locally, so I am opening the PR directly.
- We can normalize the documentation for the rest of FSDP in later follow-ups.

Differential Revision: [D35431794](https://our.internmc.facebook.com/intern/diff/D35431794)

[ghstack-poisoned]

awgu commented:
I am marking ready for review so that docs builds are not skipped, but I need to double check the docs render as expected.

awgu commented:
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431794).

awgu commented:
Returns in the docstrings look fixed:
https://docs-preview.pytorch.org/75327/fsdp.html?highlight=fullyshardeddataparallel#torch.distributed.fsdp.FullyShardedDataParallel

rohan-varma reviewed (approved):
thanks for fixing the docs!

awgu commented:
@pytorchbot merge this please
